CMakeCommon.cmake

```diff
@@ -99,8 +99,103 @@ function (LinkGSLibrariesToProject target acVersion devKitDir)
 
 endfunction ()
 
-function (GenerateAddOnProject target acVersion devKitDir addOnName addOnSourcesFolder addOnResourcesFolder addOnLanguage)
+function (ParseVersion inValue outList)
+    set (v1 0)
+    set (v2 0)
+    set (v3 0)
+    unset (CMAKE_MATCH_COUNT)
+    if (inValue MATCHES [[^([0-9]+)\.([0-9]+)\.([0-9]+)$]])
+    elseif (inValue MATCHES [[^([0-9]+)\.([0-9]+)$]])
+    elseif (inValue MATCHES [[^([0-9]+)$]])
+    endif ()
+    if (DEFINED CMAKE_MATCH_COUNT)
+        foreach (i RANGE 1 "${CMAKE_MATCH_COUNT}")
+            set ("v${i}" "${CMAKE_MATCH_${i}}")
+        endforeach ()
+        set ("${outList}" "${v1};${v2};${v3}" PARENT_SCOPE)
+    else ()
+        unset ("${outList}" PARENT_SCOPE)
+    endif ()
+endfunction ()
+
+function (GenerateAddOnVersionInfo target)
+    get_target_property (devKitDir "${target}" GSdevKitDir)
+    get_target_property (addOnName "${target}" GSaddOnName)
+    get_target_property (acVersion "${target}" GSacVersion)
+    if (NOT DEFINED devKitDir OR NOT DEFINED addOnName OR NOT DEFINED acVersion)
+        message (FATAL_ERROR "Target '${target}' was not created with GenerateAddOnProject.")
+    endif ()
+
+    cmake_parse_arguments (PARSE_ARGV 1 vers "" VERSION "")
+    if (DEFINED vers_KEYWORDS_MISSING_VALUES AND "VERSION" IN_LIST vers_KEYWORDS_MISSING_VALUES)
+        message (FATAL_ERROR "\
+'VERSION' argument is missing its value. Please make sure you replaced the placeholder in CMakeLists.txt:
+
+    GenerateAddOnVersionInfo (CMakeTarget VERSION #[[version goes here]])
+
+Replace the '#[[version goes here]]' comment with a version number that you wish see in the file's properties and bug reports.
+Accepted version number formats are:
+
+    123
+    1.23
+    1.2.3")
+    endif ()
+
+    ParseVersion ("${vers_VERSION}" vers)
+    if (NOT DEFINED vers)
+        message (FATAL_ERROR "'${vers_VERSION}' does not follow the '1.2.3' version format, where the 2nd and 3rd components are optional.")
+    endif ()
+
+    set (company "GRAPHISOFT SE")
+    string (TIMESTAMP copyright "Copyright © ${company}, 1984-%Y")
```

Comment:
The company should always be taken from the environment or the config file because these tools are part of the recommended public toolset and pipeline. Please try to modify the script.